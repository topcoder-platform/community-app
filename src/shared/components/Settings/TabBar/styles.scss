@import "~styles/mixins";

nav.settings-tab {
  display: flex;
  flex-direction: row;
  justify-content: center;
  border-bottom: 1px solid $tc-gray-20;
  background-color: $tc-gray-neutral-light;

  @media only screen and (min-width: 768px) {
    display: flex;
    flex-direction: row;
    height: 60px;
    justify-content: center;
    align-items: center;
  }

  li {
    display: inline-block;
    margin-right: 25px;

    @media only screen and (min-width: 768px) {
      margin-right: 35px;
    }

    &:last-child {
      margin-right: 0;
    }
  }

  a {
    @include sofia-pro-medium;

    font-size: 13px;
    line-height: 24px;
    color: $tc-gray-40;
    text-transform: uppercase;
    text-decoration: none;
    cursor: pointer;

    @media only screen and (min-width: 768px) {
      font-size: 16px;
    }

    &:hover {
      color: $tc-gray-90;
    }

    &:focus {
      outline: none;
    }
  }

  .active-tab {
    padding-bottom: 18px;
    border-bottom: 2px solid $tc-dark-blue-110;
    color: $tc-gray-90;
  }
}
