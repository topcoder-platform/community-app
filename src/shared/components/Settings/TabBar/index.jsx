/**
 * Child component of Settings/TabBar renders bar for 4 tabs.
 */
import React from 'react';
import PT from 'prop-types';

import { TABS } from 'actions/page/settings';
import './styles.scss';

export default function TabBar(props) {
  const {
    settingsTab,
    selectTab,
  } = props;

  const clickTab = (e, tab) => {
    e.preventDefault();
    setImmediate(() => {
      selectTab(tab);
    });
  };

  return (
    <nav styleName="settings-tab">
      <ul>
        <li><a role="link" tabIndex={0} onKeyPress={e => clickTab(e, TABS.PROFILE)} onClick={e => clickTab(e, TABS.PROFILE)} styleName={settingsTab === TABS.PROFILE ? 'active-tab' : ''}>Profile</a></li>
        <li><a role="link" tabIndex={0} onKeyPress={e => clickTab(e, TABS.ACCOUNT)} onClick={e => clickTab(e, TABS.ACCOUNT)} styleName={settingsTab === TABS.ACCOUNT ? 'active-tab' : ''}>Account</a></li>
        <li><a role="link" tabIndex={0} onKeyPress={e => clickTab(e, TABS.EMAIL)} onClick={e => clickTab(e, TABS.EMAIL)} styleName={settingsTab === TABS.EMAIL ? 'active-tab' : ''}>Email</a></li>
        <li><a role="link" tabIndex={0} onKeyPress={e => clickTab(e, TABS.PREFERENCES)} onClick={e => clickTab(e, TABS.PREFERENCES)} styleName={settingsTab === TABS.PREFERENCES ? 'active-tab' : ''}>Preferences</a></li>
      </ul>
    </nav>
  );
}

TabBar.propTypes = {
  settingsTab: PT.string.isRequired,
  selectTab: PT.func.isRequired,
};

