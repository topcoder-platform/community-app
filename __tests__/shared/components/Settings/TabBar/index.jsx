import React from 'react';
import Renderer from 'react-test-renderer/shallow';

import TabBar from 'components/Settings/TabBar';

const rnd = new Renderer();

it('renders tab bar correctly', () => {
  rnd.render((<TabBar
    settingsTab="profile"
    selectTab={() => {}}
  />));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
