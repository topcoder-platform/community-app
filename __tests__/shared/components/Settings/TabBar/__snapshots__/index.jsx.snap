// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tab bar correctly 1`] = `
<nav
  className="src-shared-components-Settings-TabBar-___styles__settings-tab___3G_Kf"
>
  <ul>
    <li>
      <a
        className="src-shared-components-Settings-TabBar-___styles__active-tab___2CNZI"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex={0}
      >
        Profile
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex={0}
      >
        Account
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex={0}
      >
        Email
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex={0}
      >
        Preferences
      </a>
    </li>
  </ul>
</nav>
`;
